lms/djangoapps/course_api/serializers.py

```diff
@@ -0,0 +1,79 @@
+"""
+Course API Serializers.  Representing course catalog data
+"""
+
+import urllib
+
+from django.core.urlresolvers import reverse
+from django.template import defaultfilters
+
+from rest_framework import serializers
+
+from xmodule.course_module import DEFAULT_START_DATE
+from lms.djangoapps.courseware.courses import (
+    course_image_url,
+    get_course_about_section,
+)
+
+
+class CourseSerializer(serializers.Serializer):  # pylint: disable=abstract-method
+    """
+    Serializer for Course objects
+    """
+
+    id = serializers.CharField(read_only=True)  # pylint: disable=invalid-name
+    name = serializers.CharField(source='display_name_with_default')
+    number = serializers.CharField(source='display_number_with_default')
+    org = serializers.CharField(source='display_org_with_default')
+    description = serializers.SerializerMethodField()
+    course_image = serializers.SerializerMethodField()
+    start = serializers.DateTimeField()
+    start_type = serializers.SerializerMethodField()
+    start_display = serializers.SerializerMethodField()
+    end = serializers.DateTimeField()
+    enrollment_start = serializers.DateTimeField()
+    enrollment_end = serializers.DateTimeField()
+    blocks_url = serializers.SerializerMethodField()
+
+    def get_start_type(self, course):
```

Comment:
This code needs to be refactored so it can be shared between this module and the [mobile API](https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/course_metadata_utils.py).  Perhaps include it in [course_metadata_utils](https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/course_metadata_utils.py) or in [CourseOverview](https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/content/course_overviews/models.py#L25) when this code is updated to use course overview instead.